### DISCUSSION TITLE
Required checks not passing for reusable workflows

### DISCUSSION THREAD

**Discussion by vitalii-dubovoi** [community/community#29825] [Actions] [answered]
**Describe the bug**
Required checks do not pass for reusable workflows

**To Reproduce
Steps to reproduce the behavior:**

1. Create a status check for a job name (ie pipeline)
2. Create a reusable workflow with a job similar to the required check (ie. pipeline)
3. Create a new workflow using the reusable workflow

**Expected behavior**
The required status check should acknowledge the workflow run

![image](https://user-images.githubusercontent.com/103578109/185182478-60ae8bb8-7688-4882-9d65-12156c31eaf6.png)

**Comment by airtower-luna** [comment, marked as answer]
This doesn't look like a bug to me. When using a reusable workflow the job name in the caller becomes a part of the job name, as you see there. So when modifying the workflow structure you'll have to update the branch protection rule to match, just like you would if you had renamed a job.

**Comment by haadr** [comment]
I'm not sure if this is a bug or not, but there is one problem with this behavior.

If you want to have a required status for the reusable workflow `docker` called by `NodeJS Docker Images`, but calling this reusable workflow can be skipped using an if: condition, then setting the required status to `docker / docker` will prevent this required status from ever being fulfilled, even though the workflow was skipped. Settings the required status to `docker` only will make github understand that this job was skipped, but then it will never be counted as fulfilled if the reusable workflow actually runs, as then the path will be `docker / docker` and no longer match the required status that's been configured.

See also https://github.com/actions/runner/issues/1917
